nixos/modules/programs/zsh/zsh.nix

```diff
@@ -156,6 +156,11 @@ in
 
         setopt HIST_IGNORE_DUPS SHARE_HISTORY HIST_FCNTL_LOCK
 
+        # Tell zsh how to find installed completions
+        for p in ''${(z)NIX_PROFILES}; do
+          fpath+=($p/share/zsh/site-functions $p/share/zsh/$ZSH_VERSION/functions $p/share/zsh/vendor-completions)
+        done
+
         HELPDIR="${pkgs.zsh}/share/zsh/$ZSH_VERSION/help"
 
         ${optionalString cfg.enableCompletion "autoload -U compinit && compinit"}
```

Comment:
It might be better to move this after `cfg.interactiveShellInit` that way extending fpath in there also works as expected.